dali/operators/geometry/affine_transforms/combine_transforms.cc

```diff
@@ -0,0 +1,133 @@
+
+namespace dali {
+
+template <int... values>
+using dims = std::integer_sequence<int, values...>;
+
+template <typename T, int mat_dim>
+using affine_mat_t = mat<mat_dim, mat_dim, T>;
+
+DALI_SCHEMA(transforms__Combine)
+  .DocStr(R"code(Combines two or more affine transforms.)code")
+  .NumInput(2, 99)
+  .NumOutput(1)
+  .AddParent("TransformAttr");
+
+class CombineTransformsCPU : public Operator<CPUBackend> {
+ public:
+  explicit CombineTransformsCPU(const OpSpec &spec) :
+      Operator<CPUBackend>(spec),
+      reverse_order_(spec.GetArgument<bool>("reverse_order")) {
+  }
+
+  bool CanInferOutputs() const override { return true; }
+
+ protected:
+  bool SetupImpl(std::vector<OutputDesc> &output_descs,
+                 const workspace_t<CPUBackend> &ws) override {
+    assert(ws.NumInput() > 1);
+    TensorListShape<> in0_shape = ws.template InputRef<CPUBackend>(0).shape();
+    ndim_ = in0_shape[0][0];
+    nsamples_ = in0_shape.size();
+
+    DALI_ENFORCE(in0_shape.sample_dim() == 2 &&
+                 in0_shape.size() > 0 &&
+                 in0_shape[0][1] == (in0_shape[0][0] + 1),
+      make_string(
+        "The input, if provided, is expected to be a 2D tensor with dimensions "
+        "(ndim, ndim+1) representing an affine transform. Got: ", in0_shape));
+
+    for (int i = 0; i < ws.NumInput(); i++) {
+      const auto &shape = ws.template InputRef<CPUBackend>(i).shape();
+      DALI_ENFORCE(shape == in0_shape,
+        make_string("All input transforms are expected to have the same shape. Got: ",
+                    in0_shape, " and ", shape, " for the ", i, "-th input."));
+    }
+
+    output_descs.resize(1);  // only one output
+    output_descs[0].type = TypeTable::GetTypeInfo(dtype_);
+    output_descs[0].shape = uniform_list_shape(nsamples_, {ndim_, ndim_+1});
+    return true;
+  }
+
+  template <typename T>
+  void RunImplTyped(workspace_t<CPUBackend> &ws, dims<>) {
+    DALI_FAIL(make_string("Unsupported number of dimensions ", ndim_));
+  }
+
+  template <typename T, int ndim, int... ndims>
+  void RunImplTyped(workspace_t<CPUBackend> &ws, dims<ndim, ndims...>) {
+    if (ndim_ != ndim) {
+      RunImplTyped<T>(ws, dims<ndims...>());
+      return;
+    }
+
+    constexpr int mat_dim = ndim + 1;
+    auto &out = ws.template OutputRef<CPUBackend>(0);
+    out.SetLayout({});  // no layout
+    auto out_view = view<T>(out);
+
+    for (int sample_idx = 0; sample_idx < nsamples_; sample_idx++) {
+      auto mat = affine_mat_t<T, mat_dim>::identity();
+      for (int input_idx = 0; input_idx < ws.NumInput(); input_idx++) {
+        auto &in = ws.template InputRef<CPUBackend>(input_idx);
+        auto in_view = view<T>(in);
+        auto next_mat = affine_mat_t<T, mat_dim>::identity();
+        for (int i = 0, k = 0; i < ndim; i++)
+          for (int j = 0; j < ndim + 1; j++, k++)
+            next_mat(i, j) = in_view[sample_idx].data[k];
+        mat = reverse_order_ ? mat * next_mat : next_mat * mat;  // mat mul
```

Comment:
Sorry for iterative review, but I think this would be worthwhile:
```suggestion
        if (input_idx)
          mat = reverse_order_ ? mat * next_mat : next_mat * mat;  // mat mul
        else
          mat = next_mat;
```